### DISCUSSION TITLE
ESP32 Wroom 32 TWAI driver help

### DISCUSSION THREAD

**Discussion by notsolowki** [collin80/SavvyCAN#674] [General]
Is there an issue with some of the esp32s receiving messages. i wired up a tja1050 to the esp32 wroom 32 but your ESP32CAN example codes dont initialize the CAN unless i set to listen_only. in listen only mode i cant get the esp32 to pickup any of the messages.   could my issue be related to this issue?

https://github.com/espressif/esp-idf/issues/5604

**Comment by collin80** [comment]
It's probably not related as they were having troubles with the Rx side but you seem to be having troubles with the Tx side. Can you scope the Rx/Tx pins on the transceiver to see if traffic is getting through as you'd expect? Have you checked the bus termination?

**Comment by notsolowki** [comment]
Sorry it was my own configuration that was the problem.

**Comment by notsolowki** [comment]
@collin80 I do have one more minor issue, maybe you could give me your opinion on. When i first turn on the esp32 it crashes over and over a few times untill it starts working. then it continues to work. it will only crash after power on or using the reset button.

My backtrace decodes to this,
`
Decoding stack results
0x40083349: panic_abort at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_system/panic.c line 408
0x40087f45: esp_system_abort at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_system/esp_system.c line 137
0x4008ce29: __assert_func at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/newlib/assert.c line 85
0x4008aca3: xPortEnterCriticalTimeout at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/esp_hw_support/include/soc/spinlock.h line 122
0x40088c26: xQueueReceive at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/freertos/port/xtensa/include/freertos/portmacro.h line 578
0x400d7447: twai_receive at /Users/ficeto/Desktop/ESP32/ESP32S2/esp-idf-public/components/driver/twai.c line 595
0x400d1d6b: task_LowLevelRX(void*) at C:\Users\LAPTOP\Documents\Arduino\libraries\esp32_can-master\src\esp32_can_builtin.cpp line 115`


this is line 115 from your library where it happens. any idea what cause's this?

        if (twai_receive(&message, pdMS_TO_TICKS(100)) == ESP_OK)

Thanks

**Comment by notsolowki** [comment]
I would like to add that the problem happens the worst when the can transceiver is plugged in. if left plugged into io 4,5 then its almost inifinite boot loop. if i leave tja1051 disconnected during startup it wont crash. im not sure what to think.

**Comment by notsolowki** [comment]
If I start your esp32_can sketch with the tja1051 unhooked it won't crash. Most times with just the tja1051 connected it will crash a couple times on startup but only somtimes. If I leave the can bus connected to CAN H, CAN L. It'd almost an infinite bootloop. 

I know you have a lot of experience with these devices. I was wondering if maybe you had an idea what the problem was.
